arwen/contexts/output.go

```diff
@@ -285,21 +298,68 @@ func (context *outputContext) AddTxValueToAccount(address []byte, value *big.Int
 
 // GetVMOutput updates the current VMOutput and returns it
 func (context *outputContext) GetVMOutput() *vmcommon.VMOutput {
-	if context.outputState.ReturnCode == vmcommon.Ok {
-		runtime := context.host.Runtime()
-		metering := context.host.Metering()
+	runtime := context.host.Runtime()
```

Comment:
Is there a reason why we are making all these checks inside the `GetVMOutput` function?
Is it just because we need to do it after the execution of a SC call? Can't we do it in a separate step/function?
I see `GetVMOutput` as a simple getter given it's name, but it does a lot of extra stuff now

Reply:
GetVMOutput is the one actually giving the official results - so it makes sense to have a set of checks inside.

Reply:
👍